MazeFollowing Robot

NuwanJ opened the pull request.

Imesh-Isuranga committed:
Added function for ObstacleAvoidRobot to run in Maze Playground.

NuwanJ commented on src/main/java/Robots/samples/ObstacleAvoidRobot.java:
@Imesh-Isuranga Can you create a new Class for the Maze Following robot, rather than updating the `ObstacleAvoidingRobot`?

Imesh-Isuranga replied:
ok

NuwanJ commented on src/resources/config/sample_mqtt.properties:
Please don't delete this file

Imesh-Isuranga replied:
ok

NuwanJ commented on src/main/java/Robots/samples/SampleRobot.java:
```diff
@@ -17,6 +17,7 @@ public void loop() throws Exception {
         super.loop();
 
         if (state == robotState.RUN) {
+            motion.move(250,150);
```
Can you remove this line in the PR?

NuwanJ commented on src/main/java/Robots/samples/ObstacleAvoidRobot.java:
```diff
@@ -54,6 +55,33 @@ public void loop() throws Exception {
+                if(dist<=1){
+                    motion.rotateDegree((int) (defaultMoveSpeed * 0.75), 91);
+                    if(distSensor.getDistance()<=1){
+                        motion.rotateDegree((int) (defaultMoveSpeed * 0.75), -91);
+                        motion.rotateDegree((int) (defaultMoveSpeed * 0.75), -91);
```
If you need 180 deg rotation, you can give it in one function call

Imesh-Isuranga replied:
If we want -180 degree rotation we want to give it as -182degree.But u have set limitation as degree < -180.0F || degree > 180.0F.Therefore I did like that.

NuwanJ commented on src/main/java/Robots/samples/ObstacleAvoidRobot.java:
```diff
@@ -54,6 +55,33 @@ public void loop() throws Exception {
+            System.out.println("dist  3 " +distSensor.getDistance());
+
+
+            if(dist<=10){
```
It is much better if you can taken out this value, `10` as a global/class variable

NuwanJ commented on src/main/java/Robots/samples/ObstacleAvoidRobot.java:
```diff
@@ -54,6 +55,33 @@ public void loop() throws Exception {
+                        }
+                    }
+                }else{
+                    motion.move(defaultMoveSpeed, defaultMoveSpeed, 100);
```
Since the cell distances are a fixed value, can you try `motion.moveDistance()` function for it?

Imesh-Isuranga committed:
Created new MazeFollowingRobot.java

replace motion.moveDistance() instead of motion.move()

added nearThreshold variable

Imesh-Isuranga committed:
Added proximity sensor to MazeFollowingRobot class

NuwanJ committed:
Review Changes
